package/ntrip_daemon/src/ntrip_settings.c

```diff
@@ -150,24 +151,57 @@ void ntrip_stop_processes()
   }
 }
 
+/* Move to libpiksi as generic function */
+static bool file_read_value(char *file_path) {
+  bool val = false;
+
+  if (access(file_path, F_OK) == -1) {
+    /* File is not created yet, system is most likely still booting */
+    piksi_log(LOG_DEBUG, "%s doesn't exist", file_path);
+    return val;
+  }
+
+  FILE* fp = fopen(file_path, "r");
+  if (fp == NULL) {
+    piksi_log(LOG_WARNING|LOG_SBP,
+              "Failed to open %s: errno = %d",
+              file_path,
+              errno);
+    return val;
+  }
+
+  val = ('1' == fgetc(fp));
+  fclose(fp);
+
+  return val;
+}
+
 static int ntrip_notify(void *context)
 {
   (void)context;
 
+  /* If NTRIP was enabled before notify call and ntrip_enabled is still true,
+   * it means that user is trying to modify some other NTRIP related setting
+   * than ntrip_enabled -> reject and instruct. */
+  if (file_read_value(NTRIP_ENABLED_FILE_PATH) && ntrip_enabled) {
```

Comment:
I'm actually not quite sure I understand why we need to read this file as the ntrip settings loop should be the only modifier of the value. Should be ok to just check `ntrip_enabled`

Reply:
Could be that I have understood wrongly how the settings framework behaves but let me try to explain how I see it with my current understanding.

There is one case why this is necessary: If user is changing `ntrip.enable` value `False` -> `True`. My assumption is that settings framework first reads the value `True` into `ntrip_enabled` and only after that `ntrip_notify` is called. Now if we only observe `ntrip_enabled` we read the value `True`, issue this new warning and the worst: don't actually do the enabling actions.

Do you think this is correct understanding?

Reply:
I'll test with `if (ntrip_enabled)` and see the results support my explanation.

Reply:
Test result supports my understanding. `if (ntrip_enabled)` doesn't let me change `ntrip.enable` value `False` -> `True`.

Reply:
Ended up reworking the logic due to other issues. See relevant next comments.